### PR TITLE
Fix readTextureToTexelViews to handle all types of formats and viewDimensions

### PR DESCRIPTION
**Pull request by greggman**
Issue #4017

### CODE REVIEW

File: src/webgpu/shader/execution/expression/call/builtin/texture_utils.spec.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  .params(u =>
+    u
+      .combineWithParams([
+        { srcFormat: 'r8unorm', texelViewFormat: 'rgba32float' },
+        { srcFormat: 'r8sint', texelViewFormat: 'rgba32sint' },
+        { srcFormat: 'r8uint', texelViewFormat: 'rgba32uint' },
+        { srcFormat: 'rgba32float', texelViewFormat: 'rgba32float' },
+        { srcFormat: 'rgba32uint', texelViewFormat: 'rgba32uint' },
+        { srcFormat: 'rgba32sint', texelViewFormat: 'rgba32sint' },
+        { srcFormat: 'depth24plus', texelViewFormat: 'rgba32float' },
```
Comment (shrekshao) [inline src/webgpu/shader/execution/expression/call/builtin/texture_utils.spec.ts:40]: Can we add a `{ srcFormat: 'depth24plus', texelViewFormat: 'r32float' },`

File: src/webgpu/shader/execution/expression/call/builtin/texture_utils.ts [hunk 2]
```diff
@@ -2651,49 +2671,65 @@ export async function readTextureToTexelViews(
               mipLevel)`;
```
Comment (shrekshao) [inline src/webgpu/shader/execution/expression/call/builtin/texture_utils.ts:2671]: nit: mipLevel -> 0

File: src/webgpu/shader/execution/expression/call/builtin/texture_utils.ts [hunk 3]
```diff
@@ -2704,12 +2740,53 @@ export async function readTextureToTexelViews(
                     global_invocation_id.x;
           let coord = vec3u(global_invocation_id.x / uni.sampleCount, global_invocation_id.yz);
           let sampleIndex = global_invocation_id.x % uni.sampleCount;
-          let mipLevel = uni.mipLevel;
+          let mipLevel = 0;
```
Comment (shrekshao) [inline src/webgpu/shader/execution/expression/call/builtin/texture_utils.ts:2743]: I think mipLevel can be removed now.

**Review by shrekshao — APPROVED** [review]
LGTM with comments

File: src/webgpu/format_info.ts [hunk 4]
```diff
@@ -1822,12 +1822,21 @@ export function isRegularTextureFormat(format: GPUTextureFormat) {
...
 /**
- * Returns true of format is both compressed and a float format, for example 'bc6h-rgb-ufloat'.
+ * Returns true if format is both compressed and a float format, for example 'bc6h-rgb-ufloat'.
  */
 export function isCompressedFloatTextureFormat(format: GPUTextureFormat) {
   return isCompressedTextureFormat(format) && format.includes('float');
 }
 
+/**
+ * Returns true if format is uint or uint
```
[the remark this answers is not in the document]
Comment (greggman) [reply src/webgpu/format_info.ts:1832]: done

File: src/webgpu/shader/execution/expression/call/builtin/texture_utils.ts (hunk 2 above)
Comment (greggman) [reply src/webgpu/shader/execution/expression/call/builtin/texture_utils.ts:2671]: thanks for catching that!

File: src/webgpu/shader/execution/expression/call/builtin/texture_utils.ts (hunk 3 above)
Comment (greggman) [reply src/webgpu/shader/execution/expression/call/builtin/texture_utils.ts:2743]: done

File: src/webgpu/shader/execution/expression/call/builtin/texture_utils.spec.ts (hunk 1 above)
Comment (greggman) [reply src/webgpu/shader/execution/expression/call/builtin/texture_utils.spec.ts:40]: added

**Pull request merged**